src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,352 @@
+        /// <exception cref="InvalidOperationException">Thrown while setting this value due to an invalid operation.</exception>
+        public string AppId
+        {
+            get
+            {
+                string appId;
+                int ret = Interop.Stc.Rule.GetAppId(_ruleHandle, out appId);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get AppId, Error - " + (StcError)ret);
+                    return "";
+                }
+                return appId;
+            }
+            set
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+                }
+                int ret = Interop.Stc.Rule.SetAppId(_ruleHandle, value);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to set AppId, Error - " + (StcError)ret);
+                    StcErrorFactory.ThrowStcException(ret);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Set the time interval for statistics rule.
+        /// </summary>
+        /// <since_tizen> tizen_5.5 </since_tizen>
+        /// <param name="from">The beginning of the time interval.</param>
+        /// <param name="to">The end of the time interval.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void SetTimeInterval(long from, long to)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+            }
+            int ret = Interop.Stc.Rule.SetTimeInterval(_ruleHandle, from, to);
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to set time interval, Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
+            }
+        }
+
+        /// <summary>
+        /// Get the time interval for statistics rule.
+        /// </summary>
+        /// <since_tizen> tizen_5.5 </since_tizen>
+        /// <param name="from">The beginning of the time interval.</param>
+        /// <param name="to">The end of the time interval.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void GetTimeInterval(out long from, out long to)
```

Comment:
CA1021: Avoid out parameters. Either split it to 2 different methods (maybe even properties) or create a struct/class as a return type for this method, which aggregates the result.

Reply:
It is split into two properties.